Lesson_2 done

freakeno16 opened the pull request.

freakeno16 committed:
Lesson_2 done

freakeno16 committed:
Update task2_1.rb

freakeno16 committed:
Update task2_2.rb

freakeno16 committed:
Update task2_6.rb

тоже все поправил

Caveman21days commented on lesson_2/task2_2.rb:
```diff
@@ -0,0 +1,7 @@
+
+for i in 10..100 do
+  if i % 5 == 0
+    p i
```
задание не выполнено. Массив не заполняется

Caveman21days commented on lesson_2/task2_3.rb:
```diff
@@ -0,0 +1,8 @@
+abc = {}
```
это что? Задание было "3. Заполнить массив числами фибоначчи до 100"

Caveman21days commented on lesson_2/task2_4.rb:
```diff
@@ -0,0 +1,6 @@
+
+abc_hash = {}
+
+("a".."z").each_with_index { |bukva, i| abc_hash[bukva] = i if abc_array.include?(bukva) }
```
не используй транслит, нейиминг переменных должен хоть как-то соответсвовать тому, что в них. относись к задаче серьезней

Caveman21days commented on lesson_2/task2_6.rb:
```diff
@@ -0,0 +1,27 @@
+  cost = gets.to_f
+  
+  p "How many: "
+  count = gets.to_i
```
цена по заданию тоже может быть нецелым числом

Caveman21days replied:
не исправлено

Caveman21days commented on lesson_2/task2_6.rb:
```diff
@@ -0,0 +1,27 @@
+hash = {}
+value = 0
```
переменная в целом не нужна, но если хочется оставить, то надо бы назвать в соответствии с действительностью

Caveman21days commented on lesson_2/task2_6.rb:
```diff
@@ -0,0 +1,27 @@
+  hash[name.to_sym] = { cost: cost, count: count }
+
+  each_cost = cost * count
+  p each_cost 
```
тут не надо ничего выводить, а вот хэш должен содержать финальную стоимость каждого товара (я бы посоветовал сразу добавить еще один ключ в хэш)

freakeno16 committed:
Update task2_2.rb

поправил

freakeno16 committed:
Update task2_3.rb

залил решение, хз как так вышло хд

freakeno16 committed:
Update task2_4.rb

убрал транслит, гугл транслейтер в помощь. vowel - гласная буква, оказывается

freakeno16 committed:
Update task2_6.rb

мне с переменной удобно решать было, переименовал ее в final_cost, теперь красота

freakeno16 committed:
Update task2_6.rb

убрал вывод, добавил ич_кост в хэш

freakeno16 committed:
Update task2_6.rb

Caveman21days commented on lesson_2/task2_2.rb:
```diff
@@ -0,0 +1,7 @@
+
+for i in 10..100 do
+  if i % 5 == 0
+  array << i
```
тут нет отступа у array
вообще, можно написать красивее: `array << i if i % 5 == 0`

Caveman21days commented on lesson_2/task2_4.rb:
```diff
@@ -0,0 +1,6 @@
+
+abc_hash = {}
+
+("a".."z").each_with_index { |vowel, i| abc_hash[vowel] = i if abc_array.include?(vowel) }
```
👍

freakeno16 committed:
Update task2_2.rb

не понял "тут нет отступа у array"
сделал красивее)

freakeno16 committed:
Update task2_6.rb

поправил, теперь to_f

freakeno16 committed:
Update task2_2.rb
